src/components/Indicator/Indicator.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+
+import {
+  createShorthandFactory,
+  UIComponent,
+  UIComponentProps,
+  commonPropTypes,
+  customPropTypes,
+} from '../../lib'
+import { ReactProps, ShorthandValue } from '../../../types/utils'
+import Icon from '../Icon/Icon'
+
+export interface IndicatorProps extends UIComponentProps {
+  /** The indicator can point towards different directions. */
+  direction?: 'start' | 'end' | 'top' | 'bottom'
+
+  /** The indicator can show specific icon if provided. */
+  icon?: ShorthandValue
+}
+
+/**
+ * An indicator is suggesting additional content next to the element it is used in.
+ */
+class Indicator extends UIComponent<ReactProps<IndicatorProps>, any> {
+  static displayName = 'Indicator'
+
+  static create: Function
+
+  static className = 'ui-indicator'
+
+  static directionMap = {
+    end: { unicode: '25B8', rotation: -90 },
+    start: { unicode: '25C2', rotation: 90 },
+    top: { unicode: '25B4', rotation: 180 },
+    bottom: { unicode: '25BE', rotation: 0 },
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({ children: false, content: false }),
+    direction: PropTypes.oneOf(['start', 'end', 'top', 'bottom']),
+    icon: customPropTypes.itemShorthand,
+  }
+
+  static defaultProps = {
+    as: 'span',
+    direction: 'bottom',
+  }
+
+  renderComponent({ ElementType, classes, unhandledProps, rtl }) {
+    const { direction, icon, color } = this.props
+    const hexUnicode =
+      direction && Indicator.directionMap[this.getDirectionBasedOnRtl(rtl, direction)].unicode
+    const contentProps = !icon
+      ? {
+          dangerouslySetInnerHTML: {
+            __html: hexUnicode && this.isHex(hexUnicode) ? `&#x${hexUnicode};` : '',
+          },
+        }
```

Comment:
This looks very strange for me, I played with this an it seems that we don't need this at all: https://codesandbox.io/s/1zxj6x8ok3

I've pinned `react@16.3` and `react-dom@16.3` to ensure that it's supported. An event tested in IE :)
https://create-react-app-i4qwctbvw.now.sh/
![image](https://user-images.githubusercontent.com/14183168/51472435-2548e900-1d82-11e9-8d60-0be28d426494.png)

Reply:
Yep, refactored. At the start I had here string, that's why I had to do this magic, totally valid and more clear code. Thanks!